clients/src/main/java/org/apache/kafka/clients/KafkaClient.java

```diff
@@ -86,7 +86,17 @@
     List<ClientResponse> poll(long timeout, long now);
 
     /**
+     * Diconnects the connection to a particular node, if there is one.
+     * Any pending ClientRequests for this connection will receive disconnections.
+     *
+     * @param nodeId The id of the node
+     */
+    void disconnect(String nodeId);
```

Comment:
Right now `AdminClient` has the invariant that if it submits a request to `NetworkClient`, that request will eventually be returned from `NetworkClient#poll`... possibly with an error status, possibly after a timeout, but in any case returned.  If we have cases where that doesn't happen, the code becomes a lot harder to reason about.  There are certainly other ways to structure things, but I think they are more complex and error-prone.